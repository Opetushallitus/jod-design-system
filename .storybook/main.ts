import type { StorybookConfig } from '@storybook/react-vite';

const config: StorybookConfig = {
  stories: [
    '../lib/**/*.mdx',
    {
      directory: '../lib/components',
      files: '**/*.stories.@(js|jsx|mjs|ts|tsx)',
      titlePrefix: 'Components',
    },
  ],

  addons: [
    '@storybook/addon-links',
    '@storybook/addon-a11y',
    '@storybook/addon-themes',
    // '@storybook/addon-designs',
    '@chromatic-com/storybook',
    '@storybook/addon-docs',
  ],

  framework: {
    name: '@storybook/react-vite',
    options: {},
  },

  core: {
    disableTelemetry: true,
  },

  docs: {
    defaultName: 'Documentation',
  },

  typescript: {
    reactDocgen: 'react-docgen-typescript',
  },
};
export default config;
