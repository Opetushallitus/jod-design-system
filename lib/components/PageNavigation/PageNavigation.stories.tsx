import type { Meta, StoryObj } from '@storybook/react-vite';
import { ServiceVariantProvider } from '../../hooks/useServiceVariant/ServiceVariantProvider';
import { LinkComponent, MenuListProps, MenuSection } from '../NavigationMenu';
import { PageNavigation } from './PageNavigation';

const meta = {
  title: 'Navigation/PageNavigation',
  component: PageNavigation,
  tags: ['autodocs'],
  argTypes: {
    serviceVariant: {
      control: { type: 'radio' },
      options: ['yksilo', 'ohjaaja', 'tietopalvelu', 'palveluportaali'],
    },
  },
  globals: {
    backgrounds: { value: 'gray', grid: false },
  },
  parameters: {
    controls: {
      exclude: ['menuRef', 'menuSection', 'itemClassname'],
    },
  },
} satisfies Meta<typeof PageNavigation>;

export default meta;

type Story = StoryObj<typeof meta>;

const parameters = {
  design: {
    type: 'figma',
    url: 'https://www.figma.com/design/6M2LrpSCcB0thlFDaQAI2J/cx_jod_client?node-id=14026-25623',
  },
};

const DummyLink = (props: LinkComponent) => <a href="/#" {...props} />;

const labels = ['Omat sivuni', 'Asetukseni', 'Suosikkini', 'Päämääräni', 'Osaamiseni', 'Kiinnostukseni'];
const osaamiseniLabels = ['Työpaikkani', 'Koulutukseni', 'Vapaa-ajan toimintoni', 'Muut osaaminen'];

const menuSection: MenuSection = {
  title: 'Tässä osiossa',
  linkItems: labels.map((label, index) => ({
    label,
    selected: index === 0,
    LinkComponent: DummyLink,
    childItems:
      label === 'Osaamiseni'
        ? osaamiseniLabels.map((osaaminenLabel) => ({
            label: osaaminenLabel,
            LinkComponent: DummyLink,
          }))
        : undefined,
  })),
};

const DefaultRender = (props: MenuListProps) => {
  return (
    <ServiceVariantProvider value={props.serviceVariant ?? 'yksilo'}>
      <PageNavigation {...props} />
    </ServiceVariantProvider>
  );
};

export const Default: Story = {
  parameters: {
    ...parameters,
    docs: {
      description: {
        story: 'PageNavigation for internal pages, active indicator is background color.',
      },
    },
  },
  render: DefaultRender,

  args: {
    activeIndicator: 'bg',
    isNested: false,
    itemClassname: '',
    menuRef: undefined,
    menuSection,
    openSubMenuLabel: 'Avaa alivalikko',
    serviceVariant: 'yksilo',
  },
};

export const DotIndicator: Story = {
  parameters: {
    ...parameters,
    docs: {
      description: {
        story: 'PageNavigation for internal pages, active indicator is a dot in accent color.',
      },
    },
  },
  render: DefaultRender,

  args: {
    activeIndicator: 'dot',
    isNested: false,
    itemClassname: '',
    menuRef: undefined,
    menuSection,
    openSubMenuLabel: 'Avaa alivalikko',
    serviceVariant: 'yksilo',
  },
};
