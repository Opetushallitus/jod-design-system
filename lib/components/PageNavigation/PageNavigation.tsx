import { tidyClasses as tc } from '../../utils';
import { MenuList, type MenuListProps } from '../NavigationMenu';

export type PageNavigationProps = MenuListProps & {
  className?: string;
};

export const PageNavigation = (props: PageNavigationProps) => {
  return (
    <div className={tc(`ds:p-6 ds:bg-white ds:rounded-lg ${props.className}`)}>
      <MenuList hideAccentBorder={props.hideAccentBorder ?? true} {...props} />
    </div>
  );
};
