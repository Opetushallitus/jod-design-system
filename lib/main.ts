import './index.css';

export { Button } from './components/Button/Button';
export { DropdownMenu } from './components/DropdownMenu/DropdownMenu';
export { NavigationBar } from './components/NavigationBar/NavigationBar';
export { RoundButton } from './components/RoundButton/RoundButton';
