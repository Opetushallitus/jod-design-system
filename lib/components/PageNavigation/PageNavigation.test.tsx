import { render, screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import { ServiceVariantProvider } from '../../main';
import { PageNavigation, PageNavigationProps } from './PageNavigation';

const menuSection = {
  title: 'Test Menu',
  linkItems: [{ label: 'Home', selected: true }, { label: 'About' }, { label: 'Contact' }],
};

const WrappedPageNavigation = (props?: Partial<PageNavigationProps>) => (
  <ServiceVariantProvider value="yksilo">
    <PageNavigation menuSection={menuSection} openSubMenuLabel="Open submenu" {...props} />
  </ServiceVariantProvider>
);

describe('PageNavigation', () => {
  it('renders with default props', () => {
    const { container } = render(<WrappedPageNavigation />);
    expect(screen.getByText('Test Menu')).toBeInTheDocument();
    expect(screen.getByText('Home')).toBeInTheDocument();
    expect(screen.getByText('About')).toBeInTheDocument();
    expect(screen.getByText('Contact')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('applies custom className', () => {
    const { container } = render(<WrappedPageNavigation className="custom-class" />);
    expect(container.querySelector('div')).toHaveClass('custom-class');
  });

  it('hides accent border by default', () => {
    render(<WrappedPageNavigation />);
    const ul = screen.getByRole('list');
    expect(ul.className).not.toMatch(/border-l-8/);
  });

  it('shows accent border when hideAccentBorder is false', () => {
    render(<WrappedPageNavigation hideAccentBorder={false} />);
    const ul = screen.getByRole('list');
    expect(ul.className).toMatch(/border-l-8/);
  });
});
