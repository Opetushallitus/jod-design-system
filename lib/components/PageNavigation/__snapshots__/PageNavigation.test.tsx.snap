// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageNavigation > renders with default props 1`] = `
<div
  class="ds:p-6 ds:bg-white ds:rounded-lg undefined"
>
  <div>
    <span
      class="ds:text-body-sm ds:mb-5 ds:mt-2 ds:flex"
    >
      Test Menu
    </span>
    <ul
      class="ds:gap-2 ds:flex ds:flex-col"
    >
      <li
        data-list-id="Home"
      >
        <div
          class="ds:flex ds:flex-row ds:space-between ds:min-h-8 ds:gap-2 ds:ml-3 undefined"
        >
          <span
            class="ds:flex ds:items-center ds:text-[14px] ds:leading-[20px] ds:flex-1 ds:p-3"
          >
            Home
          </span>
        </div>
      </li>
      <li
        data-list-id="About"
      >
        <div
          class="ds:flex ds:flex-row ds:space-between ds:min-h-8 ds:gap-2 ds:ml-3 undefined"
        >
          <span
            class="ds:flex ds:items-center ds:text-[14px] ds:leading-[20px] ds:flex-1 ds:p-3"
          >
            About
          </span>
        </div>
      </li>
      <li
        data-list-id="Contact"
      >
        <div
          class="ds:flex ds:flex-row ds:space-between ds:min-h-8 ds:gap-2 ds:ml-3 undefined"
        >
          <span
            class="ds:flex ds:items-center ds:text-[14px] ds:leading-[20px] ds:flex-1 ds:p-3"
          >
            Contact
          </span>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
